Source/Dafny/DafnyAst.cs

```diff
@@ -4698,35 +4732,47 @@ public enum ES { NotYetComputed, Never, ConsultTypeArguments }
 
   public class TupleTypeDecl : IndDatatypeDecl
   {
-    public readonly int Dims;
+    public readonly List<bool> ArgumentGhostness;
+
+    public int Dims
+    {
+      get { return ArgumentGhostness.Count; }
+    }
+    
+    public int NonGhostDims
+    {
+      get { return ArgumentGhostness.Count(x => !x); }
+    }
```

Comment:
This can be written as 
```suggestion
    public int NonGhostDims => ArgumentGhostness.Count(x => !x);
```

in C# (and similarly for the `Dims` getter above).

Reply:
Changed, thanks.